bodhi/server/consumers/masher.py

```diff
@@ -707,27 +821,64 @@ def generate_testing_digest(self):
 
     def generate_updateinfo(self):
         self.log.info('Generating updateinfo for %s' % self.release.name)
-        uinfo = ExtendedMetadata(self.release, self.request,
-                                 self.db, self.path)
+        uinfo = UpdateInfoMetaData(self.release, self.request,
+                                   self.db, self.mash_dir)
         self.log.info('Updateinfo generation for %s complete' % self.release.name)
         return uinfo
 
     def sanity_check_repo(self):
         """Sanity check our repo.
 
+            - make sure we didn't compose a repo full of symlinks
             - sanity check our repodata
         """
-        mash_path = os.path.join(self.path, self.id)
-        self.log.info("Running sanity checks on %s" % mash_path)
+        self.log.info("Running sanity checks on %s" % self.path)
 
-        # sanity check our repodata
-        arches = os.listdir(mash_path)
+        arches = os.listdir(os.path.join(self.path, 'compose', 'Everything'))
         for arch in arches:
+            # sanity check our repodata
             try:
-                repodata = os.path.join(mash_path, arch, 'repodata')
+                if arch == 'source':
+                    repodata = os.path.join(self.path, 'compose',
+                                            'Everything', arch, 'tree', 'repodata')
+                else:
+                    repodata = os.path.join(self.path, 'compose',
+                                            'Everything', arch, 'os', 'repodata')
                 sanity_check_repodata(repodata)
-            except Exception as e:
-                self.log.error("Repodata sanity check failed!\n%s" % str(e))
+            except Exception:
+                self.log.exception("Repodata sanity check failed!")
+                raise
+
+            # make sure that pungi didn't symlink our packages
```

Comment:
I was actually never really sure why Bodhi does this symlink check (it did it for mash too) - was there some problem in the past that symlinks caused? Does it happen often? Do you think we definitely need this?

Reply:
It can create symlinks if you misconfigure pungi.
The problem is that symlinks become problematic with the syncing to /pub.
So this check is really there to make sure that Pungi wasn't misconfigured and got us a repo we can't sync out.